### DISCUSSION TITLE
Can I use onClick on Link component or it's not recommended?

### DISCUSSION THREAD

**Discussion by samuelndm** [vercel/next.js#50774] [Help] [answered]
### Summary

<img width="347" alt="image" src="https://github.com/vercel/next.js/assets/19199441/1b0c4354-4742-48bd-bd21-1d9583989748">

**Comment by devjiwonchoi** [comment]
The `Link` component provides an option for `onClick` as default as you've noticed, it's your choice to do so, but don't forget that the `Link` component represents the `a` tag in HTML, so please be aware to use the component does not spoil SEO in general.

![Screenshot 2023-06-05 at 2 15 15 PM](https://github.com/vercel/next.js/assets/120007119/c080553e-06c4-466f-82bd-dc9136b032b3)

**Comment by icyJoseph** [comment, marked as answer]
You can, but then a Component consumer could pass a function that calls `preventDefault` and that'll start a chase for a bug that might not be so obvious.

Alternatively you can limit the type of onClick callbacks to `VoidFunction` for example and rework your call to:

```tsx
<Link href="/home" onClick={() => handleSomething()}></Link>
```

Also, that onClick handler won't work if people have JS disabled, but it feels like that's lowering more and more these days.

**Comment by Uy575** [comment]
Link component serving as a a tag and provide default onclick functionality, better add button in link tag
